src/topology/algebra/mul_action2.lean

```diff
@@ -0,0 +1,162 @@
+* `is_t2_of_properly_discontinuous_smul_of_t2` : The quotient by a discontinuous group action of
+a locally compact t2 space is t2.
+
+## Tags
+
+Hausdorff, discrete group, properly discontinuous, quotient space
+
+-/
+
+open_locale topological_space
+
+open filter set
+
+local attribute [instance] mul_action.orbit_rel
+
+/-- Class `has_continuous_smul₂ Γ T` says that the scalar multiplication `(•) : Γ → T → T`
+is continuous in the second argument. We use the same class for all kinds of multiplicative
+actions, including (semi)modules and algebras.
+-/
+class has_continuous_smul₂ (Γ : Type*) (T : Type*) [topological_space T] [has_scalar Γ T]
+ : Prop :=
+(continuous_smul₂ : ∀ γ : Γ, continuous (λ x : T, γ • x))
+
+/-- Class `has_continuous_vadd₂ Γ T` says that the additive action `(+ᵥ) : Γ → T → T`
+is continuous in the second argument. We use the same class for all kinds of additive actions,
+including (semi)modules and algebras.
+-/
+class has_continuous_vadd₂ (Γ : Type*) (T : Type*) [topological_space T]
+  [has_vadd Γ T] : Prop :=
+(continuous_vadd₂ : ∀ γ : Γ, continuous (λ x : T, γ +ᵥ x))
+
+attribute [to_additive has_continuous_vadd₂] has_continuous_smul₂
+
+export has_continuous_smul₂ (continuous_smul₂)
+
+export has_continuous_vadd₂ (continuous_vadd₂)
+
+/-- Class `properly_discontinuous_smul Γ T` says that the scalar multiplication `(•) : Γ → T → T`
+is properly discontinuous, that is, for any pair of compact sets `K, L` in `T`, only finitely many
+`γ:Γ` move `K` to have nontrivial intersection with `L`.
+-/
+class properly_discontinuous_smul (Γ : Type*) (T : Type*) [topological_space T]
+  [has_scalar Γ T] : Prop :=
+(finite_disjoint_inter_image : ∀ {K L : set T}, is_compact K → is_compact L →
+  set.finite {γ : Γ | (((•) γ) '' K) ∩ L ≠ ∅ })
+
+/-- Class `properly_discontinuous_vadd Γ T` says that the additive action `(+ᵥ) : Γ → T → T`
+is properly discontinuous, that is, for any pair of compact sets `K, L` in `T`, only finitely many
+`γ:Γ` move `K` to have nontrivial intersection with `L`.
+-/
+class properly_discontinuous_vadd (Γ : Type*) (T : Type*) [topological_space T]
+  [has_vadd Γ T] : Prop :=
+(finite_disjoint_inter_image : ∀ {K L : set T}, is_compact K → is_compact L →
+  set.finite {γ : Γ | (((+ᵥ) γ) '' K) ∩ L ≠ ∅ })
+
+attribute [to_additive] properly_discontinuous_smul
+
+variables {Γ : Type*} [group Γ] {T : Type*} [topological_space T] [mul_action Γ T]
+
+/-- A finite group action is always properly discontinuous
+-/
+@[priority 100, to_additive] instance fintype.properly_discontinuous_smul [fintype Γ] :
+  properly_discontinuous_smul Γ T :=
+{ finite_disjoint_inter_image := λ _ _ _ _, set.finite.of_fintype _}
+
+export properly_discontinuous_smul (finite_disjoint_inter_image)
+
+export properly_discontinuous_vadd (finite_disjoint_inter_image)
+
+/-- Scalar multiplication by an element of a group `Γ` acting on `T` is a homeomorphism from `T`
+to itself. -/
```

Comment:
This docstring is slightly misleading because it seems to describe a lemma and not a definition. Our usual way to fix that is to write "as a homeomorphism" or reorganize the sentence, starting with "The homeomorphism given by scalar multiplication by a given element of...". The same comment applies to the additive version of course.